packages/fuels-contract/src/contract.rs

```diff
@@ -465,13 +467,31 @@ where
             .collect();
 
         match self.contract_call.variable_outputs {
-            Some(ref mut outputs) => outputs.extend(new_outputs),
-            None => self.contract_call.variable_outputs = Some(new_outputs),
+            Some(ref mut outputs) => outputs.extend(new_variable_outputs),
+            None => self.contract_call.variable_outputs = Some(new_variable_outputs),
         }
 
         self
     }
 
+    /// Appends `num` `Output::Message`s to the transaction.
+    /// Note that this is a builder method, i.e. use it as a chain:
+    /// `my_contract_instance.my_method(...).add_message_outputs(num).call()`.
+    pub fn append_message_outputs(mut self, num: u64) -> Self {
+        let new_message_outputs: Vec<Output> = (0..num)
+            .map(|_| Output::Message {
+                recipient: Address::zeroed(),
+                amount: 0,
+            })
+            .collect();
```

Comment:
Another way, if you like it:
```suggestion

        let new_message_outputs = vec![
            Output::Message {
                recipient: Address::zeroed(),
                amount: 0,
            };
            num as usize
        ];
```

or this:
```rust
        let new_message_outputs = std::iter::repeat(Output::Message {
            recipient: Address::zeroed(),
            amount: 0,
        })
        .take(num as usize)
        .collect();
```